oximeter/collector/src/agent.rs

```diff
@@ -827,12 +826,10 @@ async fn resolve_nexus_with_backoff(
     };
     let do_lookup = || async {
         resolver
-            .lookup_ipv6(ServiceName::Nexus)
+            .lookup_socket_v6(ServiceName::Nexus)
             .await
             .map_err(|e| BackoffError::transient(e.to_string()))
-            .map(|ip| {
-                SocketAddr::V6(SocketAddrV6::new(ip, NEXUS_INTERNAL_PORT, 0, 0))
-            })
+            .map(|addr| SocketAddr::V6(addr))
```

Comment:
Maybe worth adding a `lookup_socket` that returns a `SocketAddr`?

Reply:
almost tempted, looked at other uses of `lookup_socket_v6` and it turns out they do almost the same thing as we're about to do here: format the `SocketAddrV6` into something like `http://{}`. so instead i'll make this match other use sites and keep it as `SocketAddrV6` the whole way.

(i'd like `lookup_socket` a lot, but it'd be backed by `lookup_service_targets` which is specifically ipv6-only. so the implication that `lookup_socket` could work for SRV records referencing ipv4 targets would be misleading)